library/manageiq.py

```diff
@@ -180,13 +191,43 @@ def generate_endpoints(self, hostname, port, token, h_hostname, h_port):
                                                      'auth_key': token}})
         return endpoints
 
+    def delete_provider(self, provider_name):
+        """ Deletes the provider
+
+        Returns:
+            the delete task id if a task was generated, whether or not
+            a change took place and a short message describing the operation
+            executed
+        """
+        message = ""
+        task_id = None
+        provider_id = self.find_provider_by_name(provider_name)
+        if provider_id:
+            try:
+                result = self.client.post('{providers_url}/{id}'.format(providers_url=self.providers_url, id=provider_id),
+                                          action='delete')
+                if result['success']:
+                    message = result['message']
+                    task_id = result['task_id']
+                    self.changed = True
+                else:
+                    message = "Failed to delete {provider_name} provider".format(provider_name=provider_name)
```

Comment:
do we get any more info from API when result['success'] is false but there wasn't exception?